### ISSUE TITLE
Classic block content not displayed if there is a YouTube embed in a classic block (wp.shortcode is not a constructor)

### ISSUE THREAD

**Issue by hexagongirl** [WordPress/gutenberg#12678]
Having just upgraded to WP 5.0, post content is not loading in the Gutenberg Classic block for one blog post (other blogs posts loading fine). To start with it was blank; if I switch to code editor, I can view the content, then when I switch back to visual, I get 'This block has encountered an error and cannot be previewed'.

Console error:
media-editor.min.js?ver=5.0:1 
Uncaught TypeError: wp.shortcode is not a constructor

There are no shortcodes in the post, just headings, list items, links and a YouTube embed.

I'm using Chrome 70 on Windows 10. The theme is a custom theme based on _s. Please let me know if I can provide more info.

**Comment by hexagongirl** [comment]
Sorry - forgot to add, yes the YouTube embed is just a plain link on it's own line.

**Comment by petenelson** [comment]
I also just ran into this issue. At the bottom of the post content was a YouTube URL
`<p>https://youtu.be/dQw4w9WgXcQ</p>`

I updated the post content without the YouTube URL and was able to edit the post, which I then manually added the video as an embed block.
```<!-- wp:core-embed/youtube {"url":"https://youtu.be/Zm2zNJvHLeE","type":"video","providerNameSlug":"youtube","className":"wp-embed-aspect-16-9 wp-has-aspect-ratio"} -->
<figure class="wp-block-embed-youtube wp-block-embed is-type-video is-provider-youtube wp-embed-aspect-16-9 wp-has-aspect-ratio"><div class="wp-block-embed__wrapper">
https://youtu.be/dQw4w9WgXcQ
</div></figure>
<!-- /wp:core-embed/youtube -->```

**Comment by designsimply** [comment]
@petenelson thanks for the note about the workaround you found. Quick question, do you have any of the same active plugins which are listed above?

I tested and am not seeing the problem but may not have the exact same environment for testing that you are working from. I suspect either a strange issue with the content or a conflict with a plugin or theme. Would it be possible for any of you to temporarily deactivate all plugins and check to see if a post you were having trouble with before works normally or if it still doesn't load properly?

**Comment by petenelson** [comment]
@designsimply I'll try to do some testing on a local site that has no plugins installed. This was on a large production site with several plugins, but none of the ones listed above.

**Comment by petenelson** [comment]
I tested again locally against a 5.0.2 install with some other plugins enabled and did not have any issues. I manually updated the content of a default post with a YouTube link, rather than creating a new post with blocks.

![image](https://user-images.githubusercontent.com/1757075/50367795-3ebf0100-0548-11e9-99ca-b8e953ca0d41.png)

The site we're having an issue on is WP 4.9.9 and GB 4.7.1. I was able to recreate the issue using the same steps you listed above, but I'm not able to disable all of the plugins since it's a large production multisite, but I have an identical local site so I'll see if I can narrow it down to one of the plugins.

**Comment by petenelson** [comment]
Definitely reproduced it locally with Shortcake Version 0.7.3. 

Our new site is Gutenberg, but the Shortcake plugin is network active. We can probably network deactivate it and enable it on specific sites to get around it.

**Comment by finalcircuit** [comment]
I've just installed 5.0.2 on an existing site. Posts with bare Youtube URLs display as a blank Classic block and can't be edited (also throw an error if you try to Edit As HTML). Other posts are OK. There were only three active plugins (wp-cache, UpdraftPlus and Google XML Sitemaps) and deactivating those made no difference. Currently using the Classic Editor plugin as most posts on that site have Youtube links. Happy to revert that for testing if it helps to isolate the issue.

**Comment by designsimply** [comment]
@petenelson when I install Shortcake 0.7.3 and then open any post, I see the following error but adding a YouTube link on it's own line in the classic editor still seems to work for me in testing using the Twenty Seventeen theme.

```
 Warning: Cannot modify header information - headers already sent by (output started at /srv/users/userfa5165fc/apps/userfa5165fc/public/wp-content/plugins/shortcode-ui/inc/fields/class-shortcode-ui-field-attachment.php:91) in /srv/users/userfa5165fc/apps/userfa5165fc/public/wp-includes/option.php on line 920
```

@finalcircuit thank you for the theme info! I tested Olsen Light 1.2.2 and can confirm that opening a previously created post with a YouTube embed within a classic block results in the classic block looking blank and the following error in the console:

```
TypeError: wp.shortcode is not a constructor
```

Note: a potential workaround I found is that if you click the option in the more menu to "Convert to Blocks" then the content appears.

@hexagongirl may I ask what theme you are using and whether it would be possible for you to temporarily switch to the Twenty Seventeen or Twenty Nineteen theme and also temporarily disable the Kirki Toolkit plugin and check to see if the problem goes away under those conditions?

**Comment by designsimply** [comment]
Thank you for the updated testing steps! I tested and confirmed the plugin conflict using WordPress 5.0.3 and Post Thumbnail Editor 2.4.8. 

It appears we've uncovered conflicts with:

1. Post Thumbnail Editor 2.4.8 plugin
1. Olsen Light 1.2.2 theme

I reported those issues at:

1. https://wordpress.org/support/topic/classic-block-is-blank-when-a-youtube-video-is-part-of-the-content/
1. https://wordpress.org/support/topic/classic-block-is-blank-when-a-youtube-video-is-part-of-the-content-2/

I couldn't verify the Shortcake (Shortcode UI) 0.7.3 plugin network-activated on a fresh WordPress Multisite 5.0.3 install resulted in the same problem.

I am closing this issue to defer to the plugin and theme authors for these cases. We should continue to report the issue to the plugin/theme authors if it comes up again for other plugins, or we can re-open this issue if we continue to see a large number of report here.
